src/date-picker/date-picker-dialog.jsx

```diff
@@ -193,12 +183,14 @@ const DatePickerDialog = React.createClass({
   },
 
   show() {
+    if (this.props.onShow && !this.state.open) this.props.onShow();
```

Comment:
the `!this.state.open` makes sure show (and dismiss) events are only fired once. prevents consecutive dispatch when going berserk with clicks!